os/binfmt/binfmt_unloadmodule.c

```diff
@@ -189,15 +190,15 @@ int unload_module(FAR struct binary_s *binp)
 			}
 
 #endif
-			for (int section_idx = 0; section_idx < BIN_MAX; section_idx++) {
-				binp->sections[section_idx] = NULL;
+			for (section_idx = 0; section_idx < BIN_MAX; section_idx++) {
+				binp->sections[section_idx] = (uint32_t)NULL;
```

Comment:
i think, this can be initiated with 0  rather typecast. since in the usage it has been taken care.